src/lib/encodeKeyName.ts

```diff
@@ -112,24 +107,23 @@ export const encodeDynamicKeyPart = (
       // TODO:
       throw new Error('The encoding of <intM> has not been implemented yet.');
     case 'bytes': {
-      const valueWithoutPrefix = value.replace('0x', '');
-      if (valueWithoutPrefix.length !== size * 2) {
+      if (!isHex(value)) {
         throw new Error(
-          `Wrong value: ${value} for dynamic key with type: ${type}. Value is not ${size} bytes long.`,
+          `Wrong value: ${value} for dynamic key with type: ${type}. Value is not in hex.`,
         );
       }
-
-      if (!isHex(valueWithoutPrefix)) {
+      const valueWithoutPrefix = value.replace('0x', '');
```

Comment:
Suggest using library function
https://docs.web3js.org/api/web3-eth-accounts/function/stripHexPrefix

```suggestion
      const valueWithoutPrefix = stripHexPrefix(value);
```